pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java

```diff
@@ -570,6 +572,81 @@ public static String[] split(String input, String delimiter, int limit) {
     return StringUtils.splitByWholeSeparator(input, delimiter, limit);
   }
 
+  /**
+   * @param input an input string for prefix strings generations.
+   * @param length the max length of the prefix strings for the string.
+   * @param regexChar the character for regex matching to be added to prefix strings generated. e.g. '^'
+   * @return generate an array of prefix strings of the string that are shorter than the specified length.
+   */
+  @ScalarFunction
+  public static String[] prefix(String input, int length, String regexChar) {
```

Comment:
The name of the function may lead users to think that this is equivalent to `input.substring(0, arg)`. But instead this is returning all prefixes <= given length.

Reply:
I think the idea here is valid. I can add 
prefix(String input, int length, String regexChar) for exact length prefix
prefix(String input, int minLength, int maxLength, String regexChar) for minMax length prefix 

this won't be equivalent to `input.substring(0, arg)` but it will have same style as `ngrams`

Reply:
If we are returning multiple prefixes/suffixes, then let's rename this to `prefixes/suffixes`. `prefix/suffix` should be reserved in case we need to add `input.substring(0, arg)` equivalent later.

Reply:
sure 
```
prefixes(String input, int minLength, int maxLength, String regexChar) for minMax length prefix
prefix(String input, int length, String regexChar) for exact length prefix 
```

Reply:
actually `prefix, suffix` for exact match won't be needed since there is already a substr() function we can use directly